ACPs/118-warp-signature-request/README.md

```diff
@@ -0,0 +1,81 @@
+
+Another example is ACP-75, which aims to implement acceptance proofs using Warp. The signature aggregation mechanism is not [specified](https://github.com/avalanche-foundation/ACPs/blob/main/ACPs/75-acceptance-proofs/README.md#signature-aggregation), which is a blocker for that ACP to be marked implementable.
+
+Standardizing the Warp Signature Request format by implementing it as a protocol-level (AvalancheGo) AppRequest type would simplify the implementation of ACP-75, and streamline signature aggregation for out-of-protocol services such as Warp message relayers.
+
+## Specification
+
+Subnet EVM and Coreth currently handle Warp message signature requests from peers by decoding the `AppRequest` bytes (forwarded from Avalanchego via the `AppHandler` interface) using a type registered with the message codec:
+
+```
+type MessageSignatureRequest struct {
+	MessageID ids.ID `serialize:"true"`
+}
+```
+
+`MessageID` represents the VM-agnostic Warp message ID, and could be ported as-is to the protocol. To do so, we would [define](https://github.com/ava-labs/avalanchego/blob/v1.11.10-status-removal/proto/p2p/p2p.proto) a new P2P `Message` type:
+
+```diff
+message Message {
+    ...
+    // App messages:
+    AppRequest app_request = 30;
+    AppResponse app_response = 31;
+    AppGossip app_gossip = 32;
+    AppError app_error = 34;
+    
++   // Warp signature messages:
++   MessageSignatureRequest = 35;
+}
+```
+
+Where `MessageSignatureRequest` is defined identically to Subnet EVM/Coreth:
+```
+message MessageSignature {
+    bytes message_id = 1;
+}
+```
+
+From there, the `MessageSignatureRequest` would be forwarded to the VM to retreive the signature.
+
+## Open Questions
+
+How should signature requests be forwarded to the VM? Here are a couple of candidates:
+- RPC Chain VM and Platform VM use the `Signer` [service](https://github.com/ava-labs/avalanchego/blob/v1.11.10-status-removal/proto/warp/message.proto) to issue and response to signature requests. Subnet EVM and Coreth could be modified to use this service.
+- Subnet EVM and Coreth currently [implement](https://github.com/ava-labs/subnet-evm/blob/v0.6.6/peer/network.go) `AppHandler` to directly receive other p2p messages. A `SignatureHandler` interface could be provided for VMs to implement.
+
+In addition to generic Warp message signature request handling, Subnet EVM and Coreth also implement handling for Warp messages whose payload consists solely of a block hash. This is provides first class support for state verification via Warp in the VM. The proposed `MessageSignatureRequest` p2p message type does not distinguish between Warp message payload types, so it would be left to the VM to make that distinction.
```

Comment:
```suggestion
In addition to generic Warp message signature request handling, Subnet EVM and Coreth also implement handling for Warp messages whose payload consists solely of a block hash. This provides first class support for state verification via Warp in the VM. The proposed `MessageSignatureRequest` p2p message type does not distinguish between Warp message payload types, so it would be left to the VM to make that distinction.
```